File: 2021/1/part2.c
#include <stdio.h>
#define WINDOW_SIZE 3

int main(int argc, char** argv)
{
        FILE *f;
        int i;
        int window[WINDOW_SIZE];
        int depth;
        int hits;

        f = fopen("input.txt", "r");
        if (!f) {
                return 1;
        }

        hits = 0;
        for(i = 0; (1 == fscanf(f, "%d", &depth)) ; i++) {
                if (i >= WINDOW_SIZE) {
                        if (depth > window[i % WINDOW_SIZE]) {
                                ++hits;
                        }
                }
                window[i % WINDOW_SIZE] = depth;
        }

        printf("%d\n", hits);
        fclose(f);
        return 0;
}

File: 2021/rs/1/part2.rs
use std::fs;
use std::io;
use std::io::prelude::*;

const WINDOW: usize = 3;

fn main() -> io::Result<()> {
    let file = fs::File::open("input.txt")?;
    let br = io::BufReader::new(file);

    let mut i: usize = 0;
    let mut cnt: i32 = 0;
    let mut depths: [i32; WINDOW] = [0; WINDOW];
    for line in br.lines() {
        if i >= WINDOW {
            let depth: i32 = line
                .expect("Failed to read line.")
                .trim()
                .parse()
                .expect("Failed to parse line.");
            if depth > depths[i % WINDOW] {
                cnt = cnt + 1;
            }
            depths[i % WINDOW] = depth;
        }
        i = i + 1;
    }

    println!("{}", cnt);

    Ok(())
}
